node/npm_dependencies.ts

```diff
@@ -62,34 +134,48 @@ export const vendorNPMSpecifiers = async (basePath: string, functions: string[],
   // resolution logic won't work.
   const nodePaths = [path.join(basePath, 'node_modules')]
 
+  // We need to create some files on disk, which we don't want to write to the
+  // project directory. If a custom directory has been specified, which happens
+  // only in tests, we use it. Otherwise, create a random temporary directory.
+  const temporaryDirectory = directory ? { path: directory } : await tmp.dir()
+
   // Do a first pass at bundling to gather a list of specifiers that should be
   // loaded as npm dependencies, because they either use the `npm:` prefix or
   // they are bare specifiers. We'll collect them in `specifiers`.
-  await build({
-    bundle: true,
-    entryPoints: functions,
-    nodePaths,
-    platform: 'node',
-    plugins: [getDependencyTrackerPlugin(specifiers)],
-    write: false,
-  })
+  try {
+    await build({
+      banner,
+      bundle: true,
+      entryPoints: functions,
+      logLevel: 'error',
+      nodePaths,
+      outdir: temporaryDirectory.path,
+      platform: 'node',
+      plugins: [getDependencyTrackerPlugin(specifiers, importMap.getContentsWithURLObjects(), pathToFileURL(basePath))],
+      write: false,
+    })
+  } catch (error) {
+    logger.system('Could not track dependencies in edge function:', error)
+    logger.user(
+      'An error occurred when trying to scan your Edge Functions for npm modules, which is an experimental feature. If you are loading npm modules, please share the errors above in https://ntl.fyi/edge-functions-npm. If you are not loading npm modules, you can ignore this message.',
```

Comment:
This will point to a forums post where we explain the feature and the fact that it's experimental.